flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -226,9 +230,21 @@ public static CuratorFramework startCuratorFramework(Configuration configuration
         }
 
         CuratorFramework cf = curatorFrameworkBuilder.build();
-
+        // This handler is only used to handle the error during start phase, and should be
+        // removed after start curator success.
+        UnhandledErrorListener unhandledErrorListener =
+                (message, throwable) -> {
+                    LOG.error(
+                            "Exiting process for unhandled error in start curator framework, "
+                                    + "error message: {}, exiting code: {}",
+                            message,
+                            ZOOKEEPER_FAILURE_EXIT_CODE,
+                            throwable);
+                    System.exit(ZOOKEEPER_FAILURE_EXIT_CODE);
```

Comment:
Now I just call `System.exit()` here. Another option I think we can throw exception here, let it be handled in entryPoint. What's your idea? @tillrohrmann

Reply:
We probably should use `FlinkSecurityManager.forceProcessExit`.

Reply:
A little question: How can we test the exit behavior of `FlinkSecurityManager.forceProcessExit` ?